`cli_reporter` / `typesrenderer` refactor

tandy-1000 opened the pull request:
Small refactor moving all leftover string formatting for P(Node|Sym|Type) from types.nim to typesrenderer.nim.
Removed some dead code, cleaned up some imports, reshuffled some consts & procs.
String formatting / rendering procs should not be in types.nim, and this should be fixed by this PR.

fixes #167

saem commented:
Keep it up, @tandy-1000!

BTW you can make things even better by adding missing docs for exported procs or if after the move they no longer need to be exported, remove the export flag `*`.

tandy-1000 commented:
rebased on devel.. 
edit: thanks to @haxscramper

saem commented:
I'm not sure types renderer should be in the AST package, seems misplaced. I'd like a follow-on refactor PR that fixes that, the heavy lifting being thinking about naming and placement.

saem commented:
Please double check to see if the currently exported procs etc still need exporting.

tandy-1000 commented:
> I'm not sure types renderer should be in the AST package, seems misplaced. I'd like a follow-on refactor PR that fixes that, the heavy lifting being thinking about naming and placement.

Not sure if I'll have time to execute this, but I'll try.

> Please double check to see if the currently exported procs etc still need exporting.

All done :).

saem commented:
Cool, last remaining bit based on your check list is the docs

tandy-1000 committed:
cli_reporter / typesrenderer refactor

Small refactor moving all leftover string formatting for P(Node|Sym|Type) from types.nim to typesrenderer.nim.
Removed some dead code, cleaned up some imports, reshuffled some consts & procs.
String formatting / rendering procs should not be in types.nim, and this should be fixed by this PR.

haxscramper commented:
had to make a minor fix for the `cli_reporter` logic, main work has already been approved, so

bors r+

haxscramper commented:
bors r-

haxscramper commented:
bors r+
